src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/AttributeRouteModel.cs

```diff
@@ -231,6 +231,7 @@ public static string ReplaceTokens(string template, IDictionary<string, string>
             var state = TemplateParserState.Plaintext;
 
             int? tokenStart = null;
+            int scope = 0;
```

Comment:
nit: var

(it's ASP.NET team coding standard)

Reply:
Thank you for the review, I'll add more tests and change to var. Just noticed I've a redundant scope decrement. Sorry for my brain fart..